rclcpp/include/rclcpp/node_impl.hpp

```diff
@@ -208,16 +208,46 @@ Node::register_param_change_callback(CallbackT && callback)
 }
 
 template<typename ParameterT>
+void
+Node::set_parameter_if_not_set(
+  const std::string & name,
+  const ParameterT & value)
+{
+  rclcpp::parameter::ParameterVariant parameter_variant;
+  if (!this->get_parameter(name, parameter_variant)) {
+    this->set_parameters({
+          rclcpp::parameter::ParameterVariant(name, value),
+        });
+  }
+}
+
+template<typename ParameterT>
 bool
-Node::get_parameter(const std::string & name, ParameterT & parameter) const
+Node::get_parameter(const std::string & name, ParameterT & value) const
 {
-  rclcpp::parameter::ParameterVariant parameter_variant(name, parameter);
+  rclcpp::parameter::ParameterVariant parameter_variant;
   bool result = get_parameter(name, parameter_variant);
-  parameter = parameter_variant.get_value<ParameterT>();
+  if (result) {
+    value = parameter_variant.get_value<ParameterT>();
```

Comment:
I don't know what is configured differently on Portable but this line seems to have caused a warning: http://ci.ros2.org/job/ci_windows/2463/warnings41Result/